include/settings/settings.h

```diff
@@ -245,17 +260,132 @@ void settings_dst_register(struct settings_store *cs);
  * API for handler lookup
  */
 
+/**
+ * @brief Splits given name to given pattern and next part
+ *
+ * This function checks if in the beginning of the provided name the
+ * given key may be found. The key is considered equal only if it ends with '\0'
+ * character or @ref SETTINGS_NAME_SEPARATOR.
+ *
+ * It means this would match:
+ * @code
+   settings_key_cmp("my_key/other/stuff", "my_key", NULL);
+   settings_key_cmp("my_key", "my_key", NULL);
+ * @endcode
+ *
+ * And below would not match:
+ * @code
+   settings_name_split("my_key/other/stuff", "my", NULL);
+   settings_name_split("my_key/other/stuff", "other", NULL);
+   settings_name_split("my_key/other/stuff", "my_keys", NULL);
+ * @endcode
+ *
+ * It is also possible to match bigger part of the patch:
+ * @code
+   settings_name_split("my_key/other/stuff", "my_key/other", NULL);
+   settings_name_split("my_key/other/stuff", "my_key/other/stuff", NULL);
+ * @endcode
+ *
+ * @param[in]  name The name to compare in a form of setting path
+ * @param[in]  key  The keyword to search for
+ * @param[out] next The beginning of the next name part to search
+ *                  or NULL if there is nothing more to search.
+ *                  Effectively it would point to the character after
+ *                  the @ref SETTINGS_NAME_SEPARATOR following the key found.
+ *
+ * @retval true  The comparison matched.
+ * @retval false The comparison does not matched.
+ */
+bool settings_name_split(const char *name, const char *key, const char **next);
+
+/**
+ * @brief Skip single name patch element
+ *
+ * Function skips single element.
+ * For "a/b/c" the pointer to "b/c" string would be returned.
+ *
+ * @param name The base name
```

Comment:
just a nip: I would keep doxygen parameters description convention with direction '[in],&[out]` (especially for pointers). It helps a lot with understanding at the firs glance.